integration-tests/src/test/java/gov/uk/address/api/client/AddressApiClient.java

```diff
@@ -0,0 +1,97 @@
+package gov.uk.address.api.client;
+
+import uk.gov.di.ipv.cri.common.library.client.ClientConfigurationService;
+import uk.gov.di.ipv.cri.common.library.client.HttpHeaders;
+
+import java.io.IOException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+public class AddressApiClient {
+    private final HttpClient httpClient;
+    private final ClientConfigurationService clientConfigurationService;
+    private static final String JSON_MIME_MEDIA_TYPE = "application/json";
+
+    public AddressApiClient(ClientConfigurationService clientConfigurationService) {
+        this.clientConfigurationService = clientConfigurationService;
+        this.httpClient = HttpClient.newBuilder().build();
+    }
+
+    public HttpResponse<String> sendPostCodeLookUpRequest(String sessionId, String postcode)
+            throws IOException, InterruptedException {
+        postcode = postcode.trim().replaceAll("\\s", "%20");
+        var request =
+                HttpRequest.newBuilder()
+                        .uri(
+                                new uk.gov.di.ipv.cri.common.library.util.URIBuilder(
+                                                this.clientConfigurationService
+                                                        .getPrivateApiEndpoint())
+                                        .setPath(
+                                                this.clientConfigurationService.createUriPath(
+                                                        "postcode-lookup/" + postcode))
+                                        .build())
+                        .header(HttpHeaders.ACCEPT, JSON_MIME_MEDIA_TYPE)
+                        .header("session_id", sessionId)
+                        .GET()
+                        .build();
+        return sendHttpRequest(request);
+    }
+
+    public HttpResponse<String> sendAddressRequest(String sessionId, String uprn, String postcode)
+            throws IOException, InterruptedException {
+
+        var requestBody =
```

Comment:
can we create an instance of an existing class here ?
& then serialise it using objectmapper
then if the class changes we will know about it in the client

Reply:
@tal-nagra 
- Does that mean we should add the ObjectMapper in the constructor of this class to initialise it? Or should we just use new operator to initialise the ObjectMapper class.
- Please also confirm if the class is `CanonicalAddress` that should be used here?

Reply:
Done.